File: javascript/q4.js
// Task: Fix the function to remove all duplicates without skipping elements.

function removeDuplicates(arr) {
    for (let i = 0; i < arr.length; i++) {
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[i] === arr[j]) {
                arr.splice(j, 1);
            }
        }
    }
    return arr;
}

/*
    Test case 1:
    Input: [1, 2, 3, 4, 5, 2, 3]
    Output: [1, 2, 3, 4, 5]

    Test case 2:
    Input: [1, 1, 2, 2, 3, 3, 4, 4, 5, 5]
    Output: [1, 2, 3, 4, 5]
*/

File: c/q4.c
// Task: Fix the function to remove all duplicates without skipping elements.

void removeDuplicates(int *arr, int *size)
{
    for (int i = 0; i <= *size; i++)
    {
        for (int j = i; j < *size; j++)
        {
            if (arr[i] = arr[j])
            {
                for (int k = j; k <= *size; k++)
                {
                    arr[k] = arr[k - 1];
                }
                *size++;
                j += 2;
            }
        }
    }
}

/*
Test case 1:
Input: 1 2 3 4 5 2 3 4 5
Output: 1 2 3 4 5

Test case 2:
Input: 1 2 3 4 5 1 2 3 4 5
Output: 1 2 3 4 5
*/
